Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -412,6 +412,97 @@ public override int GetHashCode()
             }
         }
 
+        /// <summary>
+        /// Helper class for managing the visibility of the gaze pointer to match windows mixed reality and HoloLens 2
+        /// When application starts, gaze pointer is visible. Then when articulate hands / motion controllers
+        /// appear, hide the gaze cursor. Whenever user says the voice wake word, make the 
+        /// gaze controller appear.
+        /// </summary>
+        private class GazePointerStateMachine : IMixedRealitySpeechHandler
+        {
+            private enum GazePointerState
+            {
+                Initial, // When the application starts up, the gaze pointer should be active
+                GazePointerActive, // Gaze pointer is active when no hands are visible, after "select" 
+                GazePointerInactive // Gaze pointer is inactive as soon as motion controller or articulated hand pointers appear
+            }
+            private GazePointerState gazePointerState = GazePointerState.Initial;
+            private bool activateGazeKeywordIsSet = false;
+
+            public bool IsGazePointerActive
+            {
+                get { return gazePointerState != GazePointerState.GazePointerInactive; }
+            }
+
+            public void UpdateState(HashSet<PointerData> pointers, GenericPointer currentGazePointer)
+            {
+                int numFarPointers = 0;
+                int numNearPointersActive = 0;
+                foreach (var p in pointers)
+                {
+                    if (p.Pointer != null)
+                    {
+                        if (p.Pointer is IMixedRealityNearPointer)
+                        {
+                            if (p.Pointer.IsInteractionEnabled)
+                            {
+                                numNearPointersActive++;
+                            }
+                        }
+                        else if (p.Pointer.BaseCursor != null 
+                            && !(p.Pointer == currentGazePointer)
+                            && p.Pointer.IsInteractionEnabled)
+                        {
+                            // We ignore the currentGazePointer here because for cases like HoloLens 1
+                            // hand input or the gamepad, we want to show the cursor still.
+                            numFarPointers++;
```

Comment:
Same as above. I can see a few cases where knowing the number of far pointers can be very useful.

Reply:
Okay, I'll expose the number of near and far active pointers as public property in FocusProvider. I won't make it part of the interface to avoid making a breaking change.